app/models/order.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class Order < ApplicationRecord
+  has_many :order_items
+  belongs_to :user
+  before_save :set_subtotal, :set_shipping, :set_total
+  validates :user_id, presence: true
+
+  def subtotal
+    order_items.collect { |oi| oi.valid? ? (oi.quantity * oi.unit_price) : 0 }.sum
+  end
+
+  def total
+    subtotal + shipping
+  end
+
+  def shipping
+    shipping = 5
```

Comment:
shipping what? It's good if the variables that measure something also indicate the unit in their name like `days_until_delivery` or `shipping_in_days`